utils/workers/Collectors.py

```diff
@@ -0,0 +1,161 @@
+from dataclasses import dataclass
+from abc import ABC, abstractmethod
+from pathlib import Path
+import os
+import logging
+import pydicom
+from utils.roi_tools import InvalidDICOMFile, InvalidROIError
+from flywheel import Client, FileEntry
+
+
+log = logging.getLogger(__name__)
+
+"""
+Collectors - 
+Process 2 of 4
+These are designed to pull roi information from flywheel.
+The ROI information can be on the session level (for dicoms) or the file level (niftis)
+They must identify where the information is, and then in the case of
+session level dicom data, they will only pull out ROI's that are associated with
+the input file provided to the gear. 
+
+Responsibilities:
+1. locate ohif metadata
+2. isolate ROI's associated with input file
+3. return curated metadata
+    
+Full process:
+1. Prep
+2. Collect
+3. Generate
+4. Convert
+
+"""
+
+class CollWorker(ABC):
```

Comment:
I think this could be translated almost exactly like how `Prepper` could be translated.

Actually since both preppers and collectors seem to depend on `file_type` only and not the particular conversion method, I'm wondering if they could be joined, and possibly even the creator.

My thought is there are currently 4 stages:

1. (prep) Prep local file               <---- Depends on file type
2. (collect) Pull ohif information     <---- Depends on file type
3. Generate (creator)                  <---- Depends on file type
4. Convert (converter)

And three of the stages depend really only on the input file type.  If you could make one generator function that instantiates a `Prepper`, `Collector`, and  `Generator` from that single file type that would be really slick